qiskit_ibm_provider/ibm_provider.py

```diff
@@ -708,18 +708,14 @@ def get_backend(
                 `project` are specified.
         """
         # pylint: disable=arguments-differ
-        if instance:
-            self._hgps.move_to_end(instance, last=False)  # type: ignore[attr-defined]
-        else:
-            self._hgps = self._initialize_hgps(self._auth_client)
-
         backends = self.backends(name, instance=instance, **kwargs)
         if len(backends) > 1:
             raise QiskitBackendNotFoundError(
                 "More than one backend matches the criteria"
             )
         if not backends:
             raise QiskitBackendNotFoundError("No backend matches the criteria")
+        backends[0].instance = instance
```

Comment:
It makes more sense to set the instance at backend initialization. Setting it here seems pretty fragile. What if people use `backends()` instead of `get_backend()`.